composeApp/src/commonMain/kotlin/org/ooni/probe/ui/results/ResultsViewModel.kt

```diff
@@ -85,8 +93,30 @@ class ResultsViewModel(
             .launchIn(viewModelScope)
 
         events
-            .filterIsInstance<Event.DeleteAllClick>()
-            .onEach { deleteAllResults() }
+            .filterIsInstance<Event.DeleteClick>()
+            .onEach {
+                if (!state.value.selectionEnabled) {
+                    deleteAllResults()
+                } else {
+
+                    val selectedIds = _state.value.results.values.flatMap { list -> // Suggestion 1.1
+                        list.filter { it.isSelected }.map { it.item.idOrThrow }
+                    }
+                    if (selectedIds.isNotEmpty()) {
+                        deleteResults(selectedIds)
+                        // Deselect all after deletion
+                        _state.update {
+                            it.copy(
+                                results = it.results.mapValues {
+                                        (_, list) ->
+                                    list.map { item -> item.copy(isSelected = false) }
+                                },
```

Comment:
You shouldn't have to do this, because those items will be deleted and removed from the list. You just need to turn off selection.